File: setup-backend-maven.js
const { spawn } = require('child_process');
const path = require('path');

console.log('\n=== Backend Maven Setup ===\n');

const javaHome = 'C:\\Environement\\Java\\jdk-17.0.5.8-hotspot';
const mavenCmd = 'C:\\Environement\\maven-3.8.6\\bin\\mvn.cmd';
const backendDir = path.join(__dirname, 'backend');

const args = ['clean', 'install', '-DskipTests', '-s', 'settings.xml', '--global-toolchains', 'toolchains.xml'];

console.log('Running Maven install...');
console.log(`Command: mvn ${args.join(' ')}`);
console.log(`Directory: ${backendDir}`);
console.log(`JAVA_HOME: ${javaHome}\n`);

const mvn = spawn(mavenCmd, args, {
    cwd: backendDir,
    env: {
        ...process.env,
        JAVA_HOME: javaHome,
        PATH: `${javaHome}\\bin;${process.env.PATH}`
    },
    stdio: 'inherit',
    shell: true
});

mvn.on('close', (code) => {
    if (code === 0) {
        console.log('\n✓ Backend setup completed successfully!');
        process.exit(0);
    } else {
        console.error(`\n✗ Backend setup failed with exit code: ${code}`);
        process.exit(code);
    }
});

mvn.on('error', (err) => {
    console.error(`\n✗ Failed to start Maven process: ${err.message}`);
    process.exit(1);
});

File: setup_backend_maven.py
#!/usr/bin/env python3
import subprocess
import os
import sys

# Set Java 17 path
JAVA17_HOME = r'C:\Environement\Java\jdk-17.0.5.8-hotspot'
MAVEN_HOME = r'C:\Environement\maven-3.8.6'

# Set environment
env = os.environ.copy()
env['JAVA_HOME'] = JAVA17_HOME
env['PATH'] = f"{JAVA17_HOME}\\bin;{MAVEN_HOME}\\bin;{env['PATH']}"

print(f'Using Java 17 from: {env["JAVA_HOME"]}')
print()

# Change to backend directory
os.chdir('backend')

try:
    print('Running: mvn clean install -DskipTests -gs settings.xml')
    print('This may take several minutes...')
    print()
    
    result = subprocess.run(
        ['mvn', 'clean', 'install', '-DskipTests', '-gs', 'settings.xml'],
        env=env,
        check=True
    )
    
    print()
    print('Backend setup completed successfully!')
    sys.exit(0)
except subprocess.CalledProcessError as e:
    print()
    print('Backend setup failed!')
    print(f'Error: {e}')
    sys.exit(1)
